spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/zookeeper/ZooKeeperStreamDefinitionRepository.java

```diff
@@ -154,8 +155,10 @@ public StreamDefinition findOne(String id) {
 			}
 			Map<String, String> map = ZooKeeperUtils.bytesToMap(bytes);
 			StreamDefinition streamDefinition = new StreamDefinition(id, map.get(DEFINITION_KEY));
-			List<ModuleDefinition> moduleDefinitions = objectReader.readValue(map.get(MODULE_DEFINITIONS_KEY));
-			streamDefinition.setModuleDefinitions(moduleDefinitions);
+			if (map.get(MODULE_DEFINITIONS_KEY) != null) {
```

Comment:
I may be wrong (which wouldn't be the first time ), wouldn't we see NPE's down stream?  i.e. it will just get thrown somewhere else.